### PR TITLE
Revisit #4048 to Springless JPA extension

### PR DESCRIPTION
**Pull request by zhfeng**
We get a chance to remove `transactionManager` in upstream `camel-jpa` and I revisit to introduce `QuarkusTransactionStrategy` and get removal of  `spring-orm` and `spring-tx` at all.

Related upstream camel PR
- https://github.com/apache/camel/pull/10954

### CODE REVIEW

File: docs/modules/ROOT/pages/reference/extensions/jpa.adoc [hunk 1]
```diff
@@ -50,8 +50,32 @@ endif::[]
...
+
+Also, it leverages https://quarkus.io/guides/transaction#programmatic-approach[Quarkus TX API] to provide `TransactionStrategy` implementation.
 
 When a single persistence unit is used, the Camel Quarkus JPA extension will automatically configure the JPA component with a
-`EntityManagerFactory` and `TransactionManager`.
+`EntityManagerFactory` and `TransactionStrategy`.
+
+[id="extensions-jpa-configuration-configuring-jpamessageidrepository"]
+=== Configuring JpaMessageIdRepository
+It needs to use `EntityManagerFactory` and `TransactionStrategy` from the CDI container to configure the `JpaMessageIdRepository`:
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/reference/extensions/jpa.adoc:62]:
```suggestion
To use `JpaMessageIdRepository` you must `@Inject` `EntityManagerFactory` and `TransactionStrategy` beans into your `RouteBuilder` and pass them to the `JpaMessageIdRepository`:
```

**Comment by jamesnetherton** [comment]
In general, I dislike having a lot of GraalVM substitutions on Camel code. If we can find ways in Camel to avoid such things in the future, then that would be my preferred way of doing things.

**Comment by zhfeng** [comment]
Thanks @jamesnetherton for your reviewings and what you dislike is somthing in `JpaComponentSubstitution`?

Yeah, I'm also thinking to do some refactors on camel-jpa.

**Comment by zhfeng** [comment]
@JiriOndrusek I'm very sorry that I have to revert your fix of JPA with https://github.com/apache/camel-quarkus/commit/256e2f9db81c0c5007e21933a5927a1a832ba0c7

**Comment by JiriOndrusek** [comment]
No problem, this was just a quock fix to make camel-main successful, Your complex change is taking care off it.

**Comment by zhfeng** [comment]
@jamesnetherton I rasie a PR on upstream camel to do some refactor on `JpaComponent` and it could be helpful for the graal substition. I think we can only add some substitions like

```java
@TargetClass(JpaEndpoint.class)
final class JpaEndpointSubstitution {
    @Alias
    private TransactionStrategy transactionStrategy;

    @Substitute
    protected EntityManagerFactory createEntityManagerFactory() {
        throw new UnsupportedOperationException("createEntityManagerFactory is not supported");
    }

    @Substitute
    public TransactionStrategy getTransactionStrategy() {
        return transactionStrategy;
    }
}

@TargetClass(JpaComponent.class)
final class JpaComponentSubstitution {
    @Substitute
    private void createTransactionStrategy() {
    }
}

@TargetClass(DefaultTransactionStrategy.class)
@Delete
final class DefaultTransactionStrategySubstitution {
}

```

**Comment by JiriOndrusek** [comment]
@zhfeng you might want to rebase to the current `camel-main` (if you want to trigger CI build again). some versions were updated and without the changes, the  CI build will fail early. (see https://github.com/apache/camel-quarkus/commit/71d0642c8e592533a8ed2cd3cbc154e56beff653)

**Comment by zhfeng** [comment]
@JiriOndrusek no problem and I still wait for the camel nightly build to include my fix of JpaComponent. Then I will take a look in the next week.

**Comment by zhfeng** [comment]
@JiriOndrusek it seems there are still some dependencies updates?

**Comment by JiriOndrusek** [comment]
> @JiriOndrusek it seems there are still some dependencies updates?

Hi, I'm checking it now.

**Comment by JiriOndrusek** [comment]
@zhfeng I updated the `camel-main` branch, would it be possible to rebase?

**Comment by zhfeng** [comment]
Sure. thanks @JiriOndrusek

**Comment by zhfeng** [comment]
The failing tests look like unrelated so I'm going to merge.

**Pull request merged**